moto/batch/models.py

```diff
@@ -1585,7 +1585,14 @@ def list_jobs(self, job_queue, job_status=None):
         return jobs
 
     def cancel_job(self, job_id, reason):
+        if job_id is None:
```

Comment:
[Both parameters are required](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/batch.html#Batch.Client.cancel_job) when calling the SDK, so the first two checks shouldn't be necessary

Reply:
You are right, they not needed. I have looked further into the behavior of the SDK and if  you call it with empty strings it raises a `ClientException`. I have updated `cancel_job` and `temrinate_job` to mimic this behaviour.